include/seahorn/Transforms/Instrumentation/CrabAnalysis.hh

```diff
@@ -0,0 +1,43 @@
+#ifndef _CRAB_ANALYSIS_HH_
+#define _CRAB_ANALYSIS_HH_
+
+#include "llvm/IR/Module.h"
+
+namespace llvm {
+class TargetLibraryInfoWrapperPass;
+} // namespace llvm
+
+namespace seadsa {
+class GlobalAnalysis;
+} // namespace seadsa
+
+namespace clam {
+class CrabBuilderManager;
+class InterGlobalClam;
+} // namespace clam
+
+namespace seahorn {
+using namespace llvm;
+
+class CrabAnalysis {
+  //// \brief crab's cfg builder manager
+  std::unique_ptr<clam::CrabBuilderManager> m_cfg_builder_man;
+  //// \brief crab instance
+  std::unique_ptr<clam::InterGlobalClam> m_crab;
+
+  /// \brief Creates a crab's cfg builder manager
+  void initCrabAnalysis(const llvm::Module &M, seadsa::GlobalAnalysis &dsa,
+                        llvm::TargetLibraryInfoWrapperPass &tliPass);
+  /// \brief Run crab analysis
+  void runCrabAnalysis();
+
+public:
+  CrabAnalysis() {}
+  void runCrabAnalysisOnModule(const llvm::Module &M,
+                               seadsa::GlobalAnalysis &dsa,
+                               llvm::TargetLibraryInfoWrapperPass &tliPass);
+  clam::InterGlobalClam &getCrab() { return *m_crab; }
```

Comment:
Sorry. For this one I think I cannot return a constant object since querying `range` by clam is not marked as a constant method. See:
https://github.com/seahorn/clam/blob/master/include/clam/ClamQueryAPI.hh#L31

Reply:
You can still add the `const` version. The compiler will choose the non-const version. 
I think the methods exposed by `ClamQueryAPI` are not marked as `const` because they modify a cache. But C++ provides the `mutable` keyword exactly for those cases. So you add the const version and I will try to make const the methods from `ClamQueryAPI` meanwhile

Reply:
Ok. I made it by using `const_cast` for now.

Reply:
No, this is not right. You keep both versions the `const` and the `non-const` and the compiler chooses the right one. Please just keep these two:

```
const clam::InterGlobalClam &getCrab() const { return *m_crab;}
clam::InterGlobalClam &getCrab()  { return *m_crab;}

```

Reply:
So don't use `const_cast`